### PR TITLE
Add codecov config

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

**Comment by chfast** [comment]
Here I extracted the coverage for syntax code (driven by syntax tests) and all tests themselves.
For syntax code only libsolidity is taken for consideration, but we can narrow the scope if physically possible.

All this can be seen in the check list as `codecov/project/syntax` and `codecov/project/tests`. The `codecov/project` has the code coverage for entire project (including syntax and tests parts).

The biggest issue is that syntax coverage cannot be seen on the codecov site. Maybe we should report that to codecov directly. They promised to add this feature long time ago.

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+coverage:
+  range: 50...100
+  status:
+    project:
+      default:
+        target: auto
+        paths: "!test/"
+      syntax:
+        target: auto
+        paths: "libsolidity/"
```
Comment (chriseth) [inline codecov.yml]: I think this can be `libsolidity/analysis`

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+codecov:
+  branch: develop
+coverage:
+  range: 50...100
```
Comment (chriseth) [inline codecov.yml]: I think we should set the lower bound to 70

**Comment by chfast** [comment]
@chriseth, I applied your suggestions.

**Review by chriseth — APPROVED** [review]

**Pull request merged**